### PR TITLE
Fix race condition on updating existing segment

### PR DESCRIPTION
**Pull request by jackjlli**
This PR fixes race condition when updating existing segment.

The race condition can happen when there are multiple request to update the same idealstate, and the last update wins.

The method `updateIdealState` in HelixHelper will check the idealstate record version, and fail the request if it doesn't match.

### CODE REVIEW

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java [hunk 1]
```diff
@@ -1796,16 +1796,16 @@ private boolean updateExistedSegment(String tableNameWithType, String segmentNam
     }
 
     // Set all partitions to online so that they load the new segment data
-    do {
-      final IdealState idealState = _helixAdmin.getResourceIdealState(_helixClusterName, tableNameWithType);
+    HelixHelper.updateIdealState(_helixZkManager, tableNameWithType, idealState -> {
```
Comment (siddharthteotia) [inline pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java:1799]: I thought this is always done using CAS where you try to update the version and if the version update fails, it implies someone beat you while updating the ideal state concurrently. In that case, you fetch the updated ideal state again and start. Were we not following this protocol?

Comment (siddharthteotia) [reply pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java:1799]: At least the TableRebalancer code follows this.

Comment (jackjlli) [reply pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java:1799]: It is following the this protocol in `updateIdealState` method. You can check the code here:
https://github.com/apache/incubator-pinot/blob/00fcb1d27f8daec48c1d87d46abfc8b60cfb32cf/pinot-common/src/main/java/org/apache/pinot/common/utils/helix/HelixHelper.java#L120

**Comment by jackjlli** [comment]
Discussed with @mcvsubbu offline. The method `updateExistedSegment` is a legacy code. We'll remove that part of code in the next release. Thus, there is no need to fix it. Closing this PR now.

**Pull request closed without merging**